app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -331,7 +332,7 @@ watch(tableVersionHash, () => {
 
 watchEffect(() => {
   // if the column definitions remain the same but there has been updates upstream ag grid doesn't know to change its row model or to fetch new data
-  if (nodeType.value != config.nodeType) {
+  if (!nodeType.value?.equals(config.nodeType)) {
```

Comment:
I think it would be more robust if the conditional doesn't run when they are both `undefined`--it works as-is, but given that this is a `watchEffect`,
an innocuous-looking change that adds a side-effect could lead to an infinite loop if the assignment doesn't ensure that the equality-check becomes false.